src/utils.ts

```diff
@@ -0,0 +1,8 @@
+// https://github.com/TylorS/typed-is-promise/blob/abf1514e1b6961adfc75765476b0debb96b2c3ae/src/index.ts
+
+export function isPromise<T>(p: any): p is Promise<T>;
+export function isPromise(p: any): p is Promise<any>;
+
+export function isPromise<T>(p: any): p is Promise<T> {
```

Comment:
Are these function overloads necessary. 
Define function as `export function isPromise<T = any>(p: any): p is Promise<T = any> {` should be enough.

Reply:
I'm far from familiar with TypeScript generics, but this looks reasonable. I had just copied it from the original library. Shall action.